File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        heaper = []

        for i in nums:
            heapq.heappush(heaper, (-i))

        num1 = -heapq.heappop(heaper)
        num2 = -heapq.heappop(heaper)

        return (num1 - 1) * (num2 - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution
{
public:
  int maxProduct(vector<int> &nums)
  {
    priority_queue<int> mh;
    for (auto curr : nums)
    {
      mh.push(curr);
    }
    int a = mh.top();
    mh.pop();
    int b = mh.top();
    return (a - 1) * (b - 1);
  }
};